packages/api/src/car.js

```diff
@@ -19,44 +23,90 @@ export async function carHead(request, env, ctx) {
 export async function carGet(request, env, ctx) {
   const cache = caches.default
   let res = await cache.match(request)
-  if (!res) {
-    const {
-      params: { cid },
-    } = request
-    // gateway does not support `carversion` yet.
-    // using it now means we can skip the cache if it is supported in the future
-    const url = new URL(`/api/v0/dag/export?arg=${cid}&carversion=1`, GATEWAY)
-    console.log(url.toString())
-    res = await fetch(url, { method: 'POST' })
-    if (!res.ok) {
-      // bail early. dont cache errors.
-      return res
-    }
-    // TODO: these headers should be upstreamed to ipfs impls
-    // without the content-disposition, firefox describes them as DMS files.
-    const headers = {
-      'Cache-Control': 'public, max-age=31536000',  // max max-age is 1 year
-      'Content-Type': 'application/car',
-      'Content-Disposition': `attachment; filename="${cid}.car"`,
-    }
-    // // compress if asked for? is it worth it?
-    // if (request.headers.get('Accept-Encoding').match('gzip')) {
-    //   headers['Content-Encoding'] = 'gzip'
-    // }
-    res = new Response(res.body, { ...res, headers })
-    ctx.waitUntil(cache.put(request, res.clone()))
+
+  if (res) {
+    return res
+  }
+
+  const {
+    params: { cid },
+  } = request
+  // gateway does not support `carversion` yet.
+  // using it now means we can skip the cache if it is supported in the future
+  const url = new URL(`/api/v0/dag/export?arg=${cid}&carversion=1`, GATEWAY)
+  console.log(url.toString())
+  res = await fetch(url, { method: 'POST' })
+  if (!res.ok) {
+    // bail early. dont cache errors.
+    return res
+  }
+  // TODO: these headers should be upstreamed to ipfs impls
+  // without the content-disposition, firefox describes them as DMS files.
+  const headers = {
+    'Cache-Control': 'public, max-age=31536000',  // max max-age is 1 year
+    'Content-Type': 'application/car',
+    'Content-Disposition': `attachment; filename="${cid}.car"`,
   }
+  // // compress if asked for? is it worth it?
+  // if (request.headers.get('Accept-Encoding').match('gzip')) {
+  //   headers['Content-Encoding'] = 'gzip'
+  // }
+  res = new Response(res.body, { ...res, headers })
+  ctx.waitUntil(cache.put(request, res.clone()))
+
   return res
 }
 
 /**
  * @param {import('./user').AuthenticatedRequest} request
+ * @param {import('./env').Env} env
  */
-export async function carPost(request, env, ctx) {
-  const { user, authToken } = request.auth
-  console.log('User ID', user._id)
-  console.log('Auth Token ID', authToken && authToken._id)
-  return new JSONResponse({})
+export async function carPost(request, env) {
+  const { authToken } = request.auth
+
+  const { headers } = request
+  const contentType = headers.get('content-type')
+
+  if (!contentType.includes('application/car')) {
+    return new Response(`${request.method} /car did not receive a car content type`, { status: 400 })
```

Comment:
We don't need to enforce correct content type at this endpoint - we shouldn't be getting any other types and 'application/car' is non-standard so it won't just be added automatically for users.